### PR TITLE
TS-110: Added @host parameter to regex_remap to allow matching on hos…

### PR DESCRIPTION
**Pull request by petarpenkov**
…t field.

See mail on @dev list, TS-110.

In addition, this PR addresses some minor issues with the plugin.

### CODE REVIEW

File: plugins/regex_remap/README [hunk 1]
```diff
@@ -53,11 +54,19 @@ to turn this off use the option 'no-query-string', e.g.
...
+Finally, to match on the host as well, use the option 'host', e.g. ::
+
+   ... @pparam=maps.reg @pparam=host
+
+With this enabled, the string that you will need to match will look like
+
+    //host/path?query=bar
+
 
 Note that the path to the plugin must be absolute, and by default it is
```
Comment (zwoop) [inline plugins/regex_remap/README]: I wonder if it's time to remove this README, now that we have good documentation on docs.trafficserver. Not necessarily for this PR, but maybe file a separate Jira/PR to just nuke this README? Duplicated docs just makes life miserable.

File: plugins/regex_remap/regex_remap.cc [hunk 2]
```diff
@@ -759,15 +761,19 @@ TSRemapNewInstance(int argc, char *argv[], void **ih, char * /* errbuf ATS_UNUSE
     } else if (strncmp(argv[i], "method", 6) == 0) {
       ri->method = true;
     } else if (strncmp(argv[i], "no-method", 9) == 0) {
-      ri->method = true;
+      ri->method = false;
```
Comment (zwoop) [inline plugins/regex_remap/regex_remap.cc]: Nice catch!

File: plugins/regex_remap/regex_remap.cc [hunk 3]
```diff
@@ -759,15 +761,19 @@ TSRemapNewInstance(int argc, char *argv[], void **ih, char * /* errbuf ATS_UNUSE
...
     } else if (strncmp(argv[i], "query-string", 12) == 0) {
       ri->query_string = true;
     } else if (strncmp(argv[i], "no-query-string", 15) == 0) {
       ri->query_string = false;
-    } else if (strncmp(argv[i], "matrix-parameters", 15) == 0) {
+    } else if (strncmp(argv[i], "matrix-parameters", 17) == 0) {
       ri->matrix_params = true;
-    } else if (strncmp(argv[i], "no-matrix-parameters", 18) == 0) {
+    } else if (strncmp(argv[i], "no-matrix-parameters", 20) == 0) {
       ri->matrix_params = false;
```
Comment (zwoop) [inline plugins/regex_remap/regex_remap.cc]: Wth, who wrote this crap. Can you put a comment on the Jira / description and also indicate that we fixed these completely broken cases as well here?

**Comment by zwoop** [comment]
Nicely done. 
👍

**Comment by zwoop** [comment]
Btw, one slight nit-pick (mostly for future reference): You should keep the Subject line on the commit to < 50 characters, otherwise (as you can see) Github does crazy line truncations.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/517/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/414/ for details.

**Pull request merged**